rust/src/applayer.rs

```diff
@@ -122,6 +122,10 @@ pub struct AppLayerTxData {
     /// STREAM_TOSERVER|STREAM_TOCLIENT: files possible in both dirs
     pub file_tx: u8,
 
+    /// The direction of the transaction. 0 for transactions that are
+    /// made up of request and replies.
+    direction: u8,
```

Comment:
This could probably be encoded into the flags below to remove the new field, but I wonder if this would be useful to have.